src/include/smash/experiment.h

```diff
@@ -1470,7 +1489,10 @@ void Experiment<Modus>::initialize_new_event(int event_number) {
   }
 
   particles_.reset();
-
+  // make sure this is initialized
+  if (modus_.is_collider())
+    nucleon_has_interacted_.assign(modus_.total_N_number(),false);
```

Comment:
Yeah, your style is a little too obfuscated for my taste - clang-format really doesn't help in my mind.  I prefer to let Emacs do the job for me, but still allowing for customised styling.  Any case, no biggy

Reply:
Just out of curiosity - why  insist on `{...}` for one-line `if`s and so on?  It seems archaic

Reply:
Because the style guide says so 😉 I believe it comes from the google c++ style guide we once settled on. (Even though I just saw that the google style guide makes this actually an exception for historical reasons.)

Also out of curiosity - what speaks against clang-format?

Reply:
It messes up my OCD tendency to write neatly formatted code which I believe is more readable.  E.g, I would write 

```c++
#include <random>
#include <iostream>
#include <iomanip>
#include <map>

namespace tools
{
  struct hist
  {
    hist(double low, double dx, int n)
      : _bins(), _dx(dx)
    {
      for (size_t i = 1; i <= 10; i++) _bins[low+i*_dx] = 0;
    }
    void operator()(double x)
    {
      if (x < _bins.begin()->first-_dx) return;

      auto ib = _bins.upper_bound(x);
      if (ib != _bins.end()) ib->second++;
    }
    std::map<double,int> _bins;
    double               _dx;
  };
}

namespace std
{
  std::ostream& operator<<(std::ostream& o, const tools::hist& h)
  {
    for (auto un : h._bins)
      o << std::fixed
	<< std::setw(12) << un.first - h._dx << " - "
	<< std::setw(12) << un.first         << " | "
	<< std::string(un.second, '*')       << std::endl;
    return o;
  }
}

int main()
{
  std::random_device               rnd{};
  std::default_random_engine       eng(rnd());
  std::normal_distribution<double> norm(0,1);
  tools::hist                      hist(-3,.6,10);                     
  
  for (size_t i = 0; i < 70; i++) hist(norm(eng));

  std::cout << hist << std::flush;

  return 0;
}
```

which `clang-format` turns into 

```c++
#include <iomanip>
#include <iostream>
#include <map>
#include <random>

namespace tools {
struct hist {
  hist(double low, double dx, int n) : _bins(), _dx(dx) {
    for (size_t i = 1; i <= 10; i++)
      _bins[low + i * _dx] = 0;
  }
  void operator()(double x) {
    if (x < _bins.begin()->first - _dx)
      return;

    auto ib = _bins.upper_bound(x);
    if (ib != _bins.end())
      ib->second++;
  }
  std::map<double, int> _bins;
  double _dx;
};
} // namespace tools

namespace std {
std::ostream &operator<<(std::ostream &o, const tools::hist &h) {
  for (auto un : h._bins)
    o << std::fixed << std::setw(12) << un.first - h._dx << " - "
      << std::setw(12) << un.first << " | " << std::string(un.second, '*')
      << std::endl;
  return o;
}
} // namespace std

int main() {
  std::random_device rnd{};
  std::default_random_engine eng(rnd());
  std::normal_distribution<double> norm(0, 1);
  tools::hist hist(-3, .6, 10);

  for (size_t i = 0; i < 70; i++)
    hist(norm(eng));

  std::cout << hist << std::flush;

  return 0;
}
```
which in my mind is a lot harder to read.

Reply:
Ok, I see. But it is also nice that you do not have to worry about formatting and just can run clang-format and have everything in one style, isn't it?

Anyway, I see your point, but just to give you(r OCD tendency) a heads-up, we need to run clang-format about the code and it will change, some of your formatting 😉

Reply:
> Just out of curiosity - why insist on {...} for one-line ifs and so on? It seems archaic

I think it's to avoid writing
```c++
if (condition)
    statement1();
    statement2();
```
when you actually meant:
```c++
if (condition) {
    statement1();
    statement2();
}
```

This also applies to one-line ifs, because they may be modified in the future to use more lines. Always using braces reduces the chances of such errors.

Reply:
Just quickly - if one uses a modern editor like Emacs (Vi?), then the editor will automatically do the proper indention for you,  To me, using something like `clang-format` is simply lazy and sloppy - hardly any automated tool can make it up for human eyes and contextualisation :-)